packages/rest/src/rest.server.ts

```diff
@@ -716,6 +717,8 @@ export class RestServer extends Context implements Server, HttpServerLike {
     if (requestContext) {
       spec = this.updateSpecFromRequest(spec, requestContext);
     }
+    const consolidationEnhancer = new ConsolidationEnhancer();
+    spec = consolidationEnhancer.modifySpec(spec);
```

Comment:
Implementing the tricky details in a new class & new source file is the right approach, I love how small the changes in RestServer are now 👏